amqp_connection_resource.c

```diff
@@ -173,7 +173,7 @@ int php_amqp_set_resource_read_timeout(amqp_connection_resource *resource, doubl
 	if (timeout == 0.) {
 		read_timeout = 0;
 	} else {
-		read_timeout = (int) (max(connection->read_timeout * 1.e+3 - .5e+3, 1.));
+		read_timeout = (int) (max(timeout * 1.e+3 - .5e+3, 1.));
```

Comment:
There's a comment just above that block.

    In Windows, setsockopt with SO_RCVTIMEO sets actual timeout
    to a value that's 500ms greater than specified value.
    Also, it's not possible to set timeout to any value below 500ms.
    Zero timeout works like it should, however.

To be honest, I did not check yet if this behavior of winsock is the same for windows 8 / 8.1 / 10, but as of windows 7 that was true.